src/libs/antares/study/parts/thermal/cluster.cpp

```diff
@@ -421,6 +456,66 @@ void Data::ThermalCluster::calculationOfSpinning()
     }
 }
 
+void Data::ThermalCluster::calculationOfMarketBidPerHourAndMarginalCostPerHour()
+{
+    if (costgeneration == Data::setManually || !prepro)
+        costGenManualCalculationOfMarketBidAndMarginalCostPerHour();
+    else // costgeneration == Data::useCostTimeseries
+        costGenTimeSeriesCalculationOfMarketBidAndMarginalCostPerHour();
+}
+
+void Data::ThermalCluster::costGenManualCalculationOfMarketBidAndMarginalCostPerHour()
+{
+    std::fill(thermalEconomicTimeSeries[0].marketBidCostPerHourTs.begin(),
+              thermalEconomicTimeSeries[0].marketBidCostPerHourTs.end(),
+              marketBidCost);
+    std::fill(thermalEconomicTimeSeries[0].marginalCostPerHourTs.begin(),
+              thermalEconomicTimeSeries[0].marginalCostPerHourTs.end(),
+              marginalCost);
+    return;
+}
+
+void Data::ThermalCluster::costGenTimeSeriesCalculationOfMarketBidAndMarginalCostPerHour()
+{
+    uint fuelCostWidth = prepro->fuelcost.width;
+    uint co2CostWidth = prepro->co2cost.width;
```

Comment:
```cpp
class ThermalCluster {
    EconomicInputData ecoInput;
};

uint fuelCostWidth = ecoInput->fuelcost.width;
```

Reply:
since the below solution is not applicable. I tried this one. 
Test-branch: 
https://github.com/rte-i/Antares_Simulator/tree/cr20-rtei-develop-test-solotion-ecoInput
Commits: 4cddd59 & 0632d5d 
Works fine. But for me, this looks like overdoing it.

Reply:
Did you mean this solution @flomnes ? 
the one below is not possible. fuelCost & co2cost need to be Matrix type.

Reply:
Yes, keep Matrix types for `fuelCost` and `co2cost`, just move them out of `prepro`